chain/client/src/stateless_validation/chunk_validator/mod.rs

```diff
@@ -282,6 +283,19 @@ impl Client {
     }
 
     fn send_state_witness_ack(&self, witness: &ChunkStateWitness) {
+        // If this validator generated the witness, drop the ack.
+        // This is because the ack mechanism is for measuring the network roundtrip time, so if we handle the ack to self,
+        // it will add small durations to the metric and lead to imprecise calculations about the network time.
+        if let Some(signer) = &self.validator_signer {
```

Comment:
Wait, this isn't possible right? We would never receive an chunk state witness from ourselves? Shouldn't we just panic/log_assert_fail if this happens?

Reply:
Oh right, this was from an earlier version. changing it to directly fail here.